### PR TITLE
Added minimum_version to JSON document metadata.

### PR DESCRIPTION
**Pull request by snorfalorpagus**
This PR adds a "minimum_version" property to JSON documents. If the version specified is greater than the installed version of Pywr, a warning is raised.

### CODE REVIEW

**Comment by snorfalorpagus** [comment]
In addition to the initial purpose of this branch, I've transitioned to using Travis' "container-based" builds. The Travis build is now done using the same system as Appveyor, i.e. using the conda build system. The result is a more standardised setup - you only have to specify new dependencies in `conda-recipe/meta.yaml` instead of in `.travis.yml` manually. Also, the build times on Travis are significantly reduced (15 minutes down to 3 minutes).

Need to squash some of these commits before merging...

**Comment by jetuk** [comment]
Is this ready now?

**Comment by snorfalorpagus** [comment]
Yes.

File: .travis.yml [hunk 1]
```diff
@@ -1,40 +1,30 @@
...
-    - "2.7"
-    - "3.4"
+sudo: false
 
 env:
-    - "PYWR_SOLVER=glpk"
-    - "PYWR_SOLVER=lpsolve"
+    matrix:
+        - CONDA_NPY=111  CONDA_PY=27
+        - CONDA_NPY=111  CONDA_PY=34
```
Comment (jetuk) [inline .travis.yml]: Should we add 3.5?

File: pywr/core.py [hunk 2]
```diff
@@ -428,6 +430,17 @@ def load(cls, data, model=None, path=None, solver=None):
         # data is a dictionary, make a copy to avoid modify the input
         data = copy.deepcopy(data)
 
+        # check minimum version
+        try:
+            minimum_version = data["metadata"]["minimum_version"]
+        except KeyError:
+            pass
```
Comment (jetuk) [inline pywr/core.py]: Warning for no minimum version?

File: .travis.yml (hunk 1 above)
Comment (snorfalorpagus) [inline .travis.yml]: I don't see why not. I don't see any benefit in testing other numpy versions though.

File: pywr/core.py (hunk 2 above)
Comment (snorfalorpagus) [inline pywr/core.py]: That should encourage users to add it. We'll have to modify all the test models.

File: .travis.yml (hunk 1 above)
Comment (jetuk) [inline .travis.yml]: Is this something we'll forget to update? How do you always use the latest version?

**Comment by jetuk** [comment]
I'll merge once it passes CI

**Pull request merged**